squirreldisk: Fix displaying of results

peret opened the pull request:
Currently the squirreldisk package is broken - it scans the directory, but does not display the results correctly (i.e., no graph and no file sizes are displayed).
The reason is that parallel-disk-usage from nixpkgs has been updated to a version that does not work with squirreldisk anymore (i.e. changed JSON output format).
This commit adds a patch that updates the frontend code to work with pdu's new JSON output format.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

marcin-serwin reviewed (changes requested):
The json change was just field rename, I've opened PR upstream that fixes this: https://github.com/adileo/squirreldisk/pull/47. ~You can add it as a patch to `frontent-build` to fix it with~

EDIT: Please vendor it instead, see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#vendoring-patches

```nix
    patches = [
      (fetchpatch {
        url = "https://github.com/adileo/squirreldisk/commit/cacf3fca9a2e4b5b56b983a4a08ee0ecd2d79645.patch";
        hash = "sha256-ft7Ym6bWmmMvtwUjFkcQZA1JgoRshV9a38JdQodnq2c=";
      })
    ];
```

peret commented:
Hmm... I still think the change from this PR should be made to the package. I.e. the original squirreldisk code comes with an embedded `pdu` binary and we should use that binary instead of the one from nixpkgs. The current version of squirreldisk works when installed from other sources, it should work the same in nixpkgs, with the pre-packaged `pdu` binary.

Basically, the reason this package is what it is in the first place, is because I didn't know about `autoPatchelfHook` and how to fix pre-packaged binaries when I first submitted it.

peret commented:
Huh, that makes sense. Thanks for taking the time to reply, I'll update the PR :+1:

peret committed:
squirreldisk: Fix displaying of results

Currently the squirreldisk package is broken - it scans the directory,
but does not display the results correctly (i.e., no graph and no file
sizes are displayed).
The reason is that parallel-disk-usage from nixpkgs has been updated to
a version that does not work with squirreldisk anymore (i.e. changed
JSON output format).
This commit adds a patch that updates the frontend code to work with
pdu's new JSON output format.

TomaSajt commented on pkgs/by-name/sq/squirreldisk/package.nix:
```diff
@@ -38,6 +38,11 @@ let
     dontInstall = true;
+
+    patches = [
+      # Update field names to work with pdu v0.11.0
```
BTW: the breaking change was in the 0.9.3 -> 0.10.0 bump

jtojnar replied:
Also, please include a link to the PR in the comment so that the status of the patch can be easily checked on updates.

TomaSajt replied:
Are we talking about a link to https://github.com/adileo/squirreldisk/pull/47

TomaSajt reviewed (approved):
LGTM

Another option: if we don't want to patch the squirreldisk application itself, we could just have a small inner derivation that specifically builds the older version of `pdu`.

Though I'd prefer the patching.
